app/viewmodels/tasklist/AboutYouTaskList.scala

```diff
@@ -91,13 +88,31 @@ class AboutYouTaskList @Inject()(verifyBusinessTaskList: VerifyBusinessTaskList,
   def leadPartnerDetailsRow(implicit profile: CurrentProfile): TaskListRowBuilder = {
     TaskListRowBuilder(
       messageKey = _ => "tasklist.aboutYou.leadPartnerDetails",
-      url =
-        _ => controllers.applicant.routes.LeadPartnerEntityController.showLeadPartnerEntityType.url,
+      url = _ => controllers.applicant.routes.LeadPartnerEntityController.showLeadPartnerEntityType.url,
       tagId = "leadPartnerDetailsRow",
-      checks =
-        scheme => Seq(scheme.partners.exists(_.exists(_.isLeadPartner))),
-      prerequisites =
-        _ => Seq(verifyBusinessTaskList.businessInfoRow)
+      checks = scheme => Seq(scheme.partners.exists(_.exists(_.isLeadPartner))),
+      prerequisites = _ => Seq(verifyBusinessTaskList.businessInfoRow)
+    )
+  }
+
+  def addressDetailsRow(implicit profile: CurrentProfile): TaskListRowBuilder = {
+    TaskListRowBuilder(
+      messageKey = _ => "tasklist.aboutYou.addressDetails",
+      url = vatScheme => resolveAddressRowUrl(vatScheme),
+      tagId = "addressDetailsRow",
+      checks = scheme => {
+        Seq(
+          Some(scheme.applicantDetails.exists(_.homeAddress.isDefined)),
+          if (scheme.applicantDetails.flatMap(_.previousAddress).exists(_.yesNo)) {
+            None
+          } else {
+            Some(scheme.applicantDetails.exists(_.previousAddress.exists(_.address.isDefined)))
+          }
+        ).flatten
+      },
+      prerequisites = scheme => Seq(
+        if (scheme.eligibilitySubmissionData.exists(_.isTransactor)) aboutYouTransactorTaskList.transactorPersonalDetailsRow else personalDetailsRow
```

Comment:
you don't need to check the transactor personal details row as a prerequisite here, this section doesn't deal with transactors.